indexing-service/src/test/java/org/apache/druid/indexing/common/task/KillUnusedSegmentsTaskTest.java

```diff
@@ -708,66 +695,71 @@ public void testKillMultipleUnusedSegmentsWithDifferentMaxUsedStatusLastUpdatedT
         )
     );
 
-    final DateTime maxUsedStatusLastUpdatedTime2 = DateTimes.nowUtc();
-
+    final DateTime lastUpdatedTime2 = DateTimes.nowUtc();
+    updateUsedStatusLastUpdated(segment2, lastUpdatedTime2);
+    updateUsedStatusLastUpdated(segment3, lastUpdatedTime2);
 
     final List<Interval> segmentIntervals = segments.stream()
                                                     .map(DataSegment::getInterval)
                                                     .collect(Collectors.toList());
 
     final Interval umbrellaInterval = JodaUtils.umbrellaInterval(segmentIntervals);
 
-
-    final KillUnusedSegmentsTask task1 =
-        new KillUnusedSegmentsTask(
-            null,
-            DATA_SOURCE,
-            umbrellaInterval,
-            null,
-            null,
-            false,
-            1,
-            10,
-            maxUsedStatusLastUpdatedTime1
-        );
+    final KillUnusedSegmentsTask task1 = new KillUnusedSegmentsTask(
+        null,
+        DATA_SOURCE,
+        umbrellaInterval,
+        null,
+        null,
+        false,
+        1,
+        10,
+        lastUpdatedTime1
+    );
 
     Assert.assertEquals(TaskState.SUCCESS, taskRunner.run(task1).get().getStatusCode());
 
-    final List<DataSegment> unusedSegments =
+    final List<DataSegment> observedUnusedSegments1 =
         getMetadataStorageCoordinator().retrieveUnusedSegmentsForInterval(
             DATA_SOURCE,
             umbrellaInterval,
             null,
             null
-            );
-
-    Assert.assertEquals(ImmutableList.of(segment2, segment3), unusedSegments);
-    Assert.assertEquals(new KillTaskReport.Stats(2, 3, 0), getReportedStats());
-
-    final KillUnusedSegmentsTask task2 =
-        new KillUnusedSegmentsTask(
-            null,
-            DATA_SOURCE,
-            umbrellaInterval,
-            null,
-            null,
-            false,
-            1,
-            10,
-            maxUsedStatusLastUpdatedTime2
         );
 
-    Assert.assertEquals(TaskState.SUCCESS, taskRunner.run(task2).get().getStatusCode());
+    Assert.assertEquals(ImmutableList.of(segment2, segment3), observedUnusedSegments1);
+    Assert.assertEquals(
+        new KillTaskReport.Stats(2, 3, 0),
+        getReportedStats()
+    );
 
-    final List<DataSegment> unusedSegments2 = getMetadataStorageCoordinator().retrieveUnusedSegmentsForInterval(
+    final KillUnusedSegmentsTask task2 = new KillUnusedSegmentsTask(
```

Comment:
Should we create a builder for this Task type inside this class?
Seems like many invocations use a lot of nulls.

Reply:
Good suggestion 👍 Refactored to use a builder class in this test.